sklearn/linear_model/_glm/glm.py

```diff
@@ -31,37 +31,47 @@
 from .._linear_loss import LinearModelLoss
 
 
-def _solve_singular(H, g):
+def _solve_singular(H, g, method="lstsq"):
```

Comment:
I decided to privately expose the switch between the 2 approaches to make it easier to investigate but as I found that neither seems to be helpful I do not know which to keep. If we can actually make use of the `indefinite_factorization` variant, I think we should unittest it individually.